tests/md_indent_test.py

```diff
@@ -0,0 +1,31 @@
+from rdm.render import md_indent
+
+
+def test_md_indent_basic():
+    input_markdown = "\n".join(["Line 1", "# Line 2", "## Line 3"])
+    expected_output_markdown = "\n".join(["Line 1", "Line 2", "# Line 3"])
+    assert md_indent(input_markdown, -1) == expected_output_markdown
+
+
+def test_md_indent_extra_trim():
+    input_markdown = "\n".join(["Line 1", "# Line 2", "## Line 3", "### Line 4"])
+    expected_output_markdown = "\n".join(["Line 1", "Line 2", "Line 3", "# Line 4"])
+    assert md_indent(input_markdown, -2) == expected_output_markdown
```

Comment:
I hadn't considered the case where you adjust the indent such that there is no longer any `#` left. Nice catch!

In this case I actually think we should throw an exception, since if we don't users may accidentally, and silently, turn headers into non-headers. I can't think of any case where you'd want to do this but doing it accidentally could be bad, especially if it was hidden in a big auto-generated document and was easy to miss. Let me know if you agree with this reasoning.

You can use a context manager that comes with pytest to test this. E.g., see https://docs.pytest.org/en/6.2.x/assert.html

Also, you may want to work through https://innolitics.com/10x/lessons/programming/intermediate-python/ if you aren't familiar with decorators and context managers --- both are very useful.

You'll probably want to mention this functionality in the documentation.

Also, I'd look through the standard python exception classes in the official python documentation to identify which exception is the right one to use.

Reply:
I would have thought you'd need to update this test after you made the code raise a UserWarning.

Reply:
That's odd. I did update that test. I'll see why it didn't update here.